### DISCUSSION TITLE
Compiling C++ AlphaZero and DQN

### DISCUSSION THREAD

**Discussion by madzai** [deepmind/open_spiel#1013] [Q&A] [answered]
Hello OpenSpiel team,

I am trying to compile the AlphaZero and DQN codes in the examples folder.

I am uisng the following link as reference:
https://github.com/deepmind/open_spiel/tree/master/open_spiel/algorithms/alpha_zero_torch

In global_variables.sh, I have changed some variables from OFF to ON. Below are some of the codes involved:
export OPEN_SPIEL_BUILD_WITH_LIBNOP="${OPEN_SPIEL_BUILD_WITH_LIBNOP:-"ON"}"
export OPEN_SPIEL_BUILD_WITH_LIBTORCH="${OPEN_SPIEL_BUILD_WITH_LIBTORCH:-"ON"}"
export OPEN_SPIEL_BUILD_WITH_LIBTORCH_DOWNLOAD_URL="${OPEN_SPIEL_BUILD_WITH_LIBTORCH_DOWNLOAD_URL:-"https://download.pytorch.org/libtorch/cpu/libtorch-cxx11-abi-shared-with-deps-1.10.1%2Bcpu.zip"}"
export OPEN_SPIEL_BUILD_WITH_TENSORFLOW_CC="${OPEN_SPIEL_BUILD_WITH_TENSORFLOW_CC:-"ON"}"

I am using Visual Studio Code to build the project with CMake. After CMake, it seems those variables are still OFF. Below are part of the CMake output:
[cmake] OPEN_SPIEL_BUILD_WITH_LIBNOP not set. Defaults to OFF
[cmake] OPEN_SPIEL_BUILD_WITH_LIBNOP: OFF 
[cmake] OPEN_SPIEL_BUILD_WITH_LIBTORCH not set. Defaults to OFF
[cmake] OPEN_SPIEL_BUILD_WITH_LIBTORCH: OFF 
[cmake] OPEN_SPIEL_BUILD_WITH_TENSORFLOW_CC not set. Defaults to OFF
[cmake] OPEN_SPIEL_BUILD_WITH_TENSORFLOW_CC: OFF 
...
[cmake] OPEN_SPIEL_ENABLE_TENSORFLOW not set. Defaults to OFF
[cmake] OPEN_SPIEL_ENABLE_TENSORFLOW: OFF 

As a result, AlphaZero and DQN examples codes are not compiled. According to the CMake files, those executable files will be generated if the variables in global_variables are set to be ON. 

Please kindly advise. Thank you.

**Comment by lanctot** [comment, marked as answer]
Oh sorry for the late reply!

The Windows installation does not load in global_variables.sh as that is a bash script (intended for Linux and MacOS only).

You would have to set those environment variables manually. (I assume there is a way to do this as part of your build in Visual Studio.)

Before that, you would also need to run the equivalent of what `install.sh` does to install both libnop and LibTorch in the proper places. This should not be too bad; they are just git clones or downloading of the .zip file, but you'd have to look into the script.

Apologies -- we have very little support in Windows and only the basic setups have been tested (and barely). I don't think I even tried any optional build parameters in Windows, and I'm not sure if anybody has. 

If you do manage to get it working, I'd really appreciate if you let us know and maybe contribute some docs to make OpenSpiel easier for everyone to use in Windows.
